filters/diag/diag.go

```diff
@@ -417,3 +457,103 @@ func (t *throttle) Response(ctx filters.FilterContext) {
 	rsp := ctx.Response()
 	rsp.Body = t.goThrottle(rsp.Body, true)
 }
+
+func (j *jitter) Name() string {
+	switch j.typ {
+	case normalDist:
+		return "normal"
+	case gaussianDist:
+		return "gaussian"
+	case randomDist:
+		return "random"
+	}
+	return "unknown"
+}
+
+func (j *jitter) CreateFilter(args []interface{}) (filters.Filter, error) {
+	var (
+		delay     time.Duration
+		interval  time.Duration
+		mu, sigma float64
+		err       error
+	)
+
+	if !(len(args) == 2 || len(args) == 4) {
+		return nil, errors.New("filter requires 2 or 4 parameters")
+	}
+
+	if delay, err = parseDuration(args[0]); err != nil {
+		return nil, fmt.Errorf("failed to parse duration delay %v: %w", args[0], err)
+	}
+
+	if interval, err = parseDuration(args[1]); err != nil {
+		return nil, fmt.Errorf("failed to parse duration interval %v: %w", args[1], err)
+	}
+
+	switch j.typ {
+	case normalDist:
+		mu = 0
+		sigma = 1
+
+	case gaussianDist:
+		if len(args) != 4 {
+			return nil, errors.New("gaussian jitter filter requires 4 parameters")
+		}
+
+		s, ok := args[2].(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert mu to string %v", args[2])
+		}
+
+		mu, err = strconv.ParseFloat(s, 32)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse float mu %v: %w", args[2], err)
+		}
+
+		s, ok = args[3].(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert sigma to string %v", args[3])
+		}
+
+		sigma, err = strconv.ParseFloat(s, 32)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse float sigma %v: %w", args[3], err)
+		}
+
+	case randomDist:
+	default:
+		return nil, errors.New("distribution does not exist")
+	}
+
+	return &jitter{
+		typ:      j.typ,
+		delay:    delay,
+		interval: interval,
+		mu:       mu,
+		sigma:    sigma,
+	}, nil
+}
+
+func (j *jitter) Request(filters.FilterContext) {
```

Comment:
Should there be some tests?

Reply:
that's why it's WIP :)

Reply:
done

Reply:
I've noticed `wip` label too late, sorry :)